suite-native/module-send/src/components/CryptoAmountInput.tsx

```diff
@@ -0,0 +1,116 @@
+import { ReactNode, RefObject } from 'react';
+import Animated, { SharedValue, useAnimatedStyle } from 'react-native-reanimated';
+import { TextInput } from 'react-native';
+
+import { Text, Input } from '@suite-native/atoms';
+import { useFormContext } from '@suite-native/forms';
+import { useField } from '@suite-native/forms';
+import { useCryptoFiatConverters } from '@suite-native/formatters';
+import { prepareNativeStyle, useNativeStyles } from '@trezor/styles';
+import { useFormatters } from '@suite-common/formatters';
+import { NetworkSymbol } from '@suite-common/wallet-config';
+import { Color } from '@trezor/theme';
+import { useDebounce } from '@trezor/react-utils';
+
+export type SendAmountInputProps = {
+    recipientIndex: number;
+    networkSymbol: NetworkSymbol;
+    inputRef: RefObject<TextInput>;
+    scaleValue: SharedValue<number>;
+    translateValue: SharedValue<number>;
+    isDisabled?: boolean;
+};
+
+export const sendAmountInputWrapperStyle = prepareNativeStyle<{ isDisabled: boolean }>(
+    (_, { isDisabled }) => ({
+        position: 'absolute',
+        top: isDisabled ? 2 : 0,
+        flex: 1,
+        width: '100%',
+    }),
+);
+
+export const SendAmountCurrencyLabelWrapper = ({
+    children,
+    isDisabled,
+}: {
+    children: ReactNode;
+    isDisabled: boolean;
+}) => {
+    const textColor: Color = isDisabled ? 'textSubdued' : 'textDefault';
+
+    return <Text color={textColor}>{children}</Text>;
+};
+
+export const sendAmountTransformer = (value: string) =>
+    value
+        .replace(/[^0-9\.]/g, '') // remove all non-numeric characters
+        .replace(/^\./g, '') // remove '.' symbol if it is not preceded by number
+        .replace(/(?<=\..*)\./g, '') // keep only first appearance of the '.' symbol
+        .replace(/(?<=^0+)0/g, ''); // remove all leading zeros except the first one
+
+export const CryptoAmountInput = ({
+    recipientIndex,
+    inputRef,
+    scaleValue,
+    translateValue,
+    networkSymbol,
+    isDisabled = false,
+}: SendAmountInputProps) => {
+    const { applyStyle } = useNativeStyles();
+    const { setValue, trigger } = useFormContext();
+    const { NetworkSymbolFormatter: formatter } = useFormatters();
+    const debounce = useDebounce();
+
+    const { onChange, onBlur, value } = useField({
+        name: `outputs.${recipientIndex}.amount`,
```

Comment:
nit: Maybe create some generator for similar names and reuse in FiatAmountInput and other places to prevent typos?

e.g.``` const getFieldName = (index: number, name: string) => `outputs.${index}.${name}` ```

Reply:
Also this could be typed by TS template strings I think

Reply:
Very good idea. This utils function might save us from many malfunctioning typos 👏. 264b35313b